dali/operators/generic/one_hot.cc

```diff
@@ -24,18 +24,38 @@ DALI_REGISTER_OPERATOR(OneHot, OneHot, CPU);
 DALI_SCHEMA(OneHot)
   .DocStr(R"code(Produces a one-hot encoding of the input.
 
-If the input is not a scalar (tensor consisting from one value per sample), the operator
-will fail.)code")
+Adds a new axis or converts scalar input into an axis of ``num_classes`` elements.
+
+For given input coordinate ``(x0, x1, ..., xn)``, and ``axis = k``, the output sample is specified as::
+
+  cls = input[x0, x1, ..., xn]
+  output[x0, x1, ..., xk-1, i, xk, ..., xn] = on_value if i == cls else off_value
```

Comment:
```suggestion
For given input coordinate ``input[0] = (x0, x1, ..., xn)``, and ``axis = k``, the output sample is specified as::

  output[0, x0, x1, ..., xk-1, i, xk, ..., xn] = on_value and off_value for all remaining output[0, ...]
```

Reply:
I explicitly mention sample here, the 0 doesn't make sense and the input is not equal to coordinate.